pkg/webhook/topic_validator.go

```diff
@@ -154,16 +153,9 @@ func (s *webhookServer) checkExistingKafkaTopicCRs(ctx context.Context,
 	clusterNamespace string, topic *banzaicloudv1alpha1.KafkaTopic) *admissionv1beta1.AdmissionResponse {
 	// check KafkaTopic in the referred KafkaCluster's namespace
 	kafkaTopicList := banzaicloudv1alpha1.KafkaTopicList{}
-	err := s.client.List(ctx, &kafkaTopicList,
-		client.MatchingFieldsSelector{
-			Selector: fields.OneTermEqualSelector("spec.name", topic.Spec.Name),
-		},
-		client.MatchingFieldsSelector{
-			Selector: fields.OneTermEqualSelector("spec.clusterRef.name", topic.Spec.ClusterRef.Name),
-		},
-	)
+	err := s.client.List(ctx, &kafkaTopicList, client.MatchingFields{"spec.name": topic.Spec.Name})
```

Comment:
Can't we add the `spec.clusterRef.name` to the list of fields to match by?

e.g.:
```
client.MatchingFields{"spec.name": topic.Spec.Name, "spec.clusterRef.name", topic.Spec.ClusterRef.Name}
```

Reply:
No we cant :( This error message happen when I tried this solution: Failed to list KafkaTopics. Error message: non-exact field matches are not supported by the cache"}

Reply:
For that to work we have to index the fields in the client cache e.g.
```
func addIndexerForKafkaTopic(ctx context.Context, newCache cache.Cache) error {
	nameIndexFunc := func(obj client.Object) []string {
		return []string{obj.(*v1alpha1.KafkaTopic).Spec.ClusterRef.Name}
	}
	err := newCache.IndexField(ctx, &v1alpha1.KafkaTopic{}, "spec.clusterRef.name", nameIndexFunc)
	if err != nil {
		return errors.WrapIfWithDetails(err, "could not setup indexer for field", "field", "spec.clusterRef.name")
	}

	namespaceIndexFunc := func(obj client.Object) []string {
		return []string{obj.(*v1alpha1.KafkaTopic).Spec.ClusterRef.Namespace}
	}
	err = newCache.IndexField(ctx, &v1alpha1.KafkaTopic{}, "spec.clusterRef.namespace", namespaceIndexFunc)
	if err != nil {
		return errors.WrapIfWithDetails(err, "could not setup indexer for field", "field", "spec.clusterRef.namespace")
	}
	return nil
}
```

Reply:
Multiple field selection is not supported. https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/cache_reader.go#L119

Reply:
@hi-im-aren I believe that's already used. The k8s `client` used by the topic validator is the same used by the controller-runtime: https://github.com/banzaicloud/koperator/blob/master/pkg/webhook/server.go#L63

https://github.com/banzaicloud/koperator/blob/master/main.go#L111

Not sure if this is related:
https://github.com/kubernetes-sigs/controller-runtime/blob/7ba3e559790c5e3543002a0d9670b4cef3ccf743/pkg/cache/internal/cache_reader.go#L192

https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/cache_reader.go#L119
but this looks as though only matching by a single field is supported.